common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
```

Comment:
As a note, we use the password provider interface to get secrets of various kinds, not just metadata. (See https://github.com/druid-io/druid/issues/3911 for similar)